elasticsearch/src/main/scala/com/stratio/crossdata/connector/elasticsearch/ElasticSearchRowConverter.scala

```diff
@@ -18,75 +18,87 @@ package com.stratio.crossdata.connector.elasticsearch
 import java.sql.Timestamp
 import java.sql.{Date => SQLDate}
 import java.text.SimpleDateFormat
+import java.util
 import java.util.Date
+
 import org.apache.spark.sql.Row
-import org.apache.spark.sql.catalyst.expressions.Attribute
-import org.apache.spark.sql.types.StructType
-import org.apache.spark.sql.types.DataType
-import org.apache.spark.sql.types.StringType
-import org.apache.spark.sql.types.IntegerType
-import org.apache.spark.sql.types.LongType
-import org.apache.spark.sql.types.DecimalType
-import org.apache.spark.sql.types.TimestampType
-import org.apache.spark.sql.types.NullType
-import org.apache.spark.sql.types.DoubleType
-import org.apache.spark.sql.types.BooleanType
-import org.apache.spark.sql.types.DateType
-import org.apache.spark.sql.types.Decimal
+import org.apache.spark.sql.catalyst.expressions.{Attribute, GenericRowWithSchema}
+import org.apache.spark.sql.types._
 import org.elasticsearch.search.SearchHit
 import org.elasticsearch.search.SearchHitField
 import org.joda.time.DateTime
 
+import scala.collection.JavaConverters._
+
 object ElasticSearchRowConverter {
 
 
   def asRows(schema: StructType, array: Array[SearchHit], requiredFields: Seq[Attribute]): Array[Row] = {
-    import scala.collection.JavaConverters._
-    val schemaMap = schema.map(field => field.name -> field.dataType).toMap
 
     array map { hit =>
-      hitAsRow(hit.fields().asScala.toMap, schemaMap, requiredFields.map(_.name))
+      hitAsRow(
+        hit.fields().asScala.toMap,
+        Option(hit.getSource).map(_.asScala.toMap).getOrElse(Map.empty),
+        schema,
+        requiredFields.map(_.name)
+      )
     }
   }
 
   def hitAsRow(
                 hitFields: Map[String, SearchHitField],
-                schemaMap: Map[String, DataType],
+                subDocuments: Map[String, AnyRef],
+                schema: StructType,
                 requiredFields: Seq[String]): Row = {
-    val values: Seq[Any] = requiredFields.map {
-      name =>
-        hitFields.get(name).flatMap(v => Option(v)).map(
-          toSQL(_, schemaMap(name))).orNull
-    }
-    Row.fromSeq(values)
-  }
 
-  def toSQL(value: SearchHitField, dataType: DataType): Any = {
+    val schemaMap = schema.map(field => field.name -> field.dataType).toMap
+
+    val values: Seq[Any] = requiredFields.map { name =>
 
-    Option(value).map { case value =>
-      //Assure value is mapped to schema constrained type.
-      enforceCorrectType(value.getValue, dataType)
-    }.orNull
-  }
+      // TODO: What if a nested subdocument is targeted
```

Comment:
Native plans are currently constrained so no queries selecting nested sub-documents can be executed using native plans. If they weren't, this code would fail. That's why I kept the comment.